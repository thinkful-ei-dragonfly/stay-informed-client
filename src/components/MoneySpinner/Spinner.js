import React from 'react';
import './Spinner.scss';
import MoneySpinner from './money-spinner.json'
import Lottie from 'react-lottie';

export default function Spinner () {

const defaultOptions = {
      loop: true,
      autoplay: true,
      animationData: MoneySpinner,
    };

  return (
    <div className='spinner-div' >
      <Lottie className='lottie-player' options={defaultOptions}/>
      <p>It takes a little while to count this much money, we'll finish in just a second.</p>
    </div>
  )
}
