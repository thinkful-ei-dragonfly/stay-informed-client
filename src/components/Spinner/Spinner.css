.spinner-div {
  text-align: center;

}
.loader {
  /* border: 16px solid #f3f3f3; 
  border-top: 16px solid #3498db; */
  /* font-size: 20px; */
  border-radius: 50%;
  width: 20px;
  height: 20px;
  animation: spin 2s linear infinite;
  color: rgb(0,159,183)
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
} 