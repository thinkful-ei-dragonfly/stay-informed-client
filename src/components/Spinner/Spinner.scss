.spinner-div {
  text-align: center;
  max-width: 500px;
  margin: 0 auto;
  p {
    text-align: center;
  }
}
