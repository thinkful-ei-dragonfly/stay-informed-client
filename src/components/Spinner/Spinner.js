import React from 'react';
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// import { library } from "@fortawesome/fontawesome-svg-core";
// import { faUser } from "@fortawesome/free-solid-svg-icons";
import './Spinner.css';
// import lottie from 'lottie-web'
import NewspaperSpinner from './newspaper-spinner.json'
import Lottie from 'react-lottie';

// library.add(faUser);

export default function Spinner () {
//   lottie.loadAnimation({
//   container: ('#animationDiv'), // the dom element that will contain the animation
//   renderer: 'svg',
//   loop: true,
//   autoplay: true,
//   path: NewspaperSpinner // the path to the animation json
// });
// <FontAwesomeIcon icon='user' />
const defaultOptions = {
      loop: true,
      autoplay: true,
      animationData: NewspaperSpinner,

    };
debugger
  return (
    <div className='spinner-div' >
      <Lottie className='lottie-player' options={defaultOptions}/>
      <p>Please wait while we find your representatives</p>
    </div>
  )
}
