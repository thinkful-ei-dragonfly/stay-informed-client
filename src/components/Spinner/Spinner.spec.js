import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Spinner from './Spinner'
import Lottie from 'react-lottie';

describe(`Spinner component`, () => {
  it('renders a Spinner', () => {
    const wrapper = shallow(<Spinner />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
