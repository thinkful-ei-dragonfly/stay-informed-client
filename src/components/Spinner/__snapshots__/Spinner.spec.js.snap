// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component renders a form.Spinner by default 1`] = `
<div
  className="spinner-div"
>
  <FontAwesomeIcon
    border={false}
    className="loader"
    fixedWidth={false}
    flip={null}
    icon="spinner"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <p>
    Please wait while we find your representatives
  </p>
</div>
`;
